### DISCUSSION TITLE
Parallel (CGAL) CDT for mesh booleans

### DISCUSSION THREAD

**Discussion by proc-sim** [libigl/libigl#2286] [General]
Hi @alecjacobson ,

The CDT stage of your mesh boolean code represents the largest bottleneck in performance. Apparently CGAL lazy evaluation is the cause, because you discussed this issue with CGAL devs a few years ago (https://github.com/CGAL/cgal/issues/2685). 

It seems like the CGAL devs were aware of this and were working on a fix/workaround, but I'm having trouble following the conversations between the various CGAL devs over the years in the many CGAL-repro threads that document this issue and its potential solutions.

Do you know if CGAL has addressed the root cause in a way that would allow the CDT stage of the boolean function to be threaded?

Thanks!

**Comment by alecjacobson** [comment]
If I remember correctly, the issue that I posted about is now been solved.
I'm still not sure if we're doing the best parallel computation in the mesh
booleans code, but I'm not sure how it should be improved either.
